cdap-ui/app/cdap/components/FileBrowser/FilePath/index.js

```diff
@@ -65,32 +75,103 @@ export default class FilePath extends Component {
     });
   }
 
-  render() {
+  renderCollapsedDropdown(collapsedLinks) {
     return (
-      <div className="file-path-container">
+      <div className="collapsed-dropdown">
+        <Dropdown
+          isOpen={this.state.showCollapsedDropdown}
+          toggle={this.toggleCollapsedDropdown}
+          className="collapsed-dropdown-toggle"
+        >
+          <DropdownToggle>
+            ...
+          </DropdownToggle>
+          <DropdownMenu>
+            {
+              collapsedLinks.map((path) => {
+                return (
+                  <DropdownItem>
+                    <Link
+                      key={path.id}
+                      to={path.link}
+                    >
+                      {path.name}
+                    </Link>
+                  </DropdownItem>
+                );
+              })
+            }
+          </DropdownMenu>
+        </Dropdown>
+      </div>
+    );
+  }
+
+  renderBreadcrumb(links) {
+    return (
+      <div className="paths">
         {
-          this.state.paths.map((path, index) => {
+          links.map((path, index) => {
             return (
               <Link
                 key={path.id}
                 to={path.link}
-                className={classnames({'active-directory': index === this.state.paths.length - 1})}
+                className={classnames({'active-directory': index === links.length - 1})}
               >
                 {path.name}
                 {
-                  index !== this.state.paths.length - 1 ? <span className="fa fa-chevron-right path-divider"></span> : null
+                  index !== links.length - 1 ? <span className="path-divider">/</span> : null
                 }
               </Link>
-
             );
           })
         }
       </div>
     );
   }
+
+  renderCollapsedView() {
+    let splitIndex = this.state.paths.length - VIEW_LIMIT + 1;
+
+    let collapsedLinks = this.state.paths.slice(0, splitIndex);
+    let displayLinks = this.state.paths.slice(splitIndex);
+
+    const collapsed = this.renderCollapsedDropdown(collapsedLinks);
+    const breadcrumb = this.renderBreadcrumb(displayLinks);
+
+    return (
+      <div className="collapsed-paths">
+        {collapsed}
+        <span className="path-divider">/</span>
+        {breadcrumb}
+      </div>
+    );
+  }
+
+  render() {
+    if (this.props.loading) {
+      return (
+        <div className="file-path-container">
```

Comment:
Not sure if we need to show spinner for the breadcrumb. Can we show one for the entire content assuming we use we have one end point for all content